tiledb/api/src/stats.rs

```diff
@@ -1,11 +1,36 @@
 use std::collections::HashMap;
+use std::ops::Deref;
 
 use anyhow::anyhow;
 use serde::{Deserialize, Serialize};
 
 use crate::error::Error;
 use crate::Result as TileDBResult;
 
+pub(crate) enum RawStatsString {
+    Owned(*mut std::ffi::c_char),
+}
+
+impl Deref for RawStatsString {
+    type Target = *mut std::ffi::c_char;
+    fn deref(&self) -> &Self::Target {
+        let RawStatsString::Owned(ref ffi) = *self;
+        ffi
+    }
+}
+
+impl Drop for RawStatsString {
+    fn drop(&mut self) {
+        let RawStatsString::Owned(ref mut ffi) = *self;
+        let res = unsafe {
+            ffi::tiledb_stats_free_str(ffi as *mut *mut std::ffi::c_char)
+        };
+        // This is currently a hardcoded return value so this assertion only
+        // exists to know if that ever changes.
+        assert!(res == ffi::TILEDB_OK);
```

Comment:
`assert_eq` instead so that we see what the result is, if it does change?  Then we'd get a nicer bug report at least.

Reply:
Done!